dgraphtest/load.go

```diff
@@ -51,77 +61,152 @@ func readGzFile(sf string) ([]byte, error) {
 		}
 	}()
 
-	gr, err := gzip.NewReader(fd)
+	data, err := readGzData(c, fd)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error creating a gzip reader for file [%v]", sf)
+		return nil, errors.Wrapf(err, "error reading data from file [%v]", sf)
+	}
+	return data, nil
+}
+
+func readGzData(c *LocalCluster, r io.Reader) ([]byte, error) {
+	if c.conf.encryption {
+		encKey, err := os.ReadFile(encKeyPath)
+		if err != nil {
+			return nil, errors.Wrap(err, "error reading the encryption key from disk")
+		}
+		r, err = enc.GetReader(encKey, r)
+		if err != nil {
+			return nil, errors.Wrap(err, "error creating encrypted reader")
+		}
+	}
+
+	gr, err := gzip.NewReader(r)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating gzip reader")
 	}
 	defer func() {
 		if err := gr.Close(); err != nil {
-			log.Printf("[WARNING] error closing gzip reader for file [%v]: %v", sf, err)
+			log.Printf("[WARNING] error closing gzip reader: %v", err)
 		}
 	}()
 
 	data, err := io.ReadAll(gr)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error reading file content [%v]", sf)
+		return nil, errors.Wrap(err, "error reading data from io.Reader")
 	}
 	return data, nil
 }
 
+func writeGzData(c *LocalCluster, data []byte) (io.Reader, error) {
```

Comment:
Could the following functions:
`writeGzData`,  `readGzData`, `setGraphQLSchema`, `findGrootAndGuardians` be methods on LocalCluster (eg `func (c *LocalCluster) LiveLoad`) ? 


More broadly any reason to have functions and methods interspersed?

Reply:
I'd have personally liked them to be methods to be more readable in some sense, But it is a personal preference tbh.